test/functorch/test_aotdispatch.py

```diff
@@ -724,6 +724,50 @@ def forward(self, primals_1):
     return [add]""",
         )
 
+    def test_input_mutation_set__into_same_input(self):
+        import torch.distributed._composable.fsdp._fsdp_param
+
+        def f(a):
+            b = torch.arange(9, dtype=a.dtype).view(3, 3)
+            c = torch.arange(9, dtype=a.dtype).view(3, 3)
+            d = torch.arange(9, dtype=a.dtype).view(3, 3)
+            with torch.no_grad(), torch.autograd._unsafe_preserve_version_counter(a):
+                torch.ops.fsdp.set_(a, b)
+            x = a * a
+            with torch.no_grad(), torch.autograd._unsafe_preserve_version_counter(a):
+                torch.ops.fsdp.set_(a, c)
+            y = a * a
+            with torch.no_grad(), torch.autograd._unsafe_preserve_version_counter(a):
+                torch.ops.fsdp.set_(a, c)
+            z = a * a
+            return x + y + z
+
+        inp = [torch.ones(3, 3, requires_grad=True)]
+        fw_graph = self.verify_aot_autograd(
+            f, inp, test_mutation=True, keep_inp_mutations=True
+        )
+        inp = [torch.ones(3, 3, requires_grad=False)]
+        self.verify_aot_autograd(f, inp, test_mutation=True, keep_inp_mutations=True)
+        self.assertExpectedInline(
```

Comment:
do you mind just mentioning the context somewhere (e.g. comment in this test):

(1) the important bit is that when there are multiple set_() calls on a graph input, they all show up with a primal as the first arg in the graph

(2) this doesn't actually work today with normall aten.set_, blocked on https://github.com/pytorch/pytorch/issues/129892, but using a custom op in fsdp with no returns is a simple workaround.